src/permission-ui/permission-ui.js

```diff
@@ -112,40 +125,37 @@ function run($rootScope, $state, PermTransitionProperties, PermTransitionEvents,
     function handleAuthorizedState() {
       PermTransitionEvents.broadcastPermissionAcceptedEvent();
 
-      // Overwrite notify option to broadcast it later
-      var transitionOptions = angular.extend({}, PermTransitionProperties.options, {notify: false, location: true});
-
-      $state
-        .go(PermTransitionProperties.toState.name, PermTransitionProperties.toParams, transitionOptions)
-        .then(function () {
-          PermTransitionEvents.broadcastStateChangeSuccessEvent();
-        });
+      return delegateTransitionTo();
     }
 
     /**
      * Handles redirection for unauthorized access
      * @method
      * @private
      *
-     * @param rejectedPermission {String} Rejected access right
      * @param statePermissionMap {permission.ui.PermPermissionMap} State permission map
+     * @returns {Function} A function that accepts the rejectedPermission access right
      */
-    function handleUnauthorizedState(rejectedPermission, statePermissionMap) {
-      PermTransitionEvents.broadcastPermissionDeniedEvent();
-
-      statePermissionMap
-        .resolveRedirectState(rejectedPermission)
-        .then(function (redirect) {
-          $state.go(redirect.state, redirect.params, redirect.options);
-        });
+    function handleUnauthorizedState(statePermissionMap) {
+      return function(rejectedPermission) {
+        PermTransitionEvents.broadcastPermissionDeniedEvent();
+
+        return statePermissionMap
```

Comment:
I'm slightly torn whether or not the original promise should resolve/reject with the transition of the redirect. If we choose not to relay the new redirect transition to the original caller, we should likely expose the newly created transition promise on the rejected error object in some way, so at least the caller could use it.